pypdf/annotations/_base.py

```diff
@@ -1,27 +1,124 @@
 from abc import ABC
+from typing import Any, Callable, Dict, Iterable, Optional
 
 from ..constants import AnnotationFlag
-from ..generic import NameObject, NumberObject
-from ..generic._data_structures import DictionaryObject
+from ..generic import (
+    ArrayObject,
+    BooleanObject,
+    DictionaryObject,
+    FloatObject,
+    NameObject,
+    NumberObject,
+    RectangleObject,
+    TextStringObject,
+)
 
+DEFAULT_FLAGS = AnnotationFlag(0)
+### !!!!!!! 0 is not actually no flags but the default value !!!!!
 
-class AnnotationDictionary(DictionaryObject, ABC):
-    def __init__(self) -> None:
-        from ..generic._base import NameObject
 
+class AnnotationDictionary(DictionaryObject, ABC):
+    def __init__(self, call_post_init: bool = False, **kwargs: Any) -> None:
+        DictionaryObject.__init__(self)
         # "rect" should not be added here as PolyLine can automatically set it
-        self[NameObject("/Type")] = NameObject("/Annot")
-        # The flags was NOT added to the constructor on purpose: We expect that
-        #   most users don't want to change the default. If they want, they
-        #   can use the property. The default is 0.
+        self._set_name("/Type", "/Annot")
+        if call_post_init:
+            self._post_init(**kwargs)
+
+    def _post_init(self, **kwargs: Any) -> None:
+        props = self.lst_properties()
+        for k, v in kwargs.items():
+            if k in props:
+                setattr(self, k, v)
+
+    def lst_properties(self) -> Dict[str, Callable]:
+        """Returns a dictionnary with the setters"""
+        return {
+            x: v.setter
+            for (x, v) in self.__class__.__dict__.items()
+            if isinstance(v, property)
+        }
 
     @property
     def flags(self) -> AnnotationFlag:
-        return self.get(NameObject("/F"), AnnotationFlag(0))
+        return self.get("/F", DEFAULT_FLAGS)
 
     @flags.setter
-    def flags(self, value: AnnotationFlag) -> None:
-        self[NameObject("/F")] = NumberObject(value)
+    def flags(
+        self, value: Optional[AnnotationFlag]
+    ) -> "AnnotationDictionary":  # to return cascading
+        self._set_int("/F", value)
+        return self
+
+    @property
+    def rect(self) -> str:
+        """
+        gets/sets the rectangle area containing the annotation
+        Changing directly this value is not recommended for Lines, Polylines, Freetext with callouts
+        """
+        return self.get("/Rect", "")
+
+    @rect.setter
+    def rect(self, value: Optional[RectangleObject]) -> "AnnotationDictionary":
+        self._set_str("/Rect", value)
+        return self
+
+    """
+    internal functions used to modify properties
+    """
+
+    def _set_bool(self, key: str, value: Optional[bool]) -> None:
+        if value is None:
+            if key in self:
+                del self[NameObject(key)]
+        else:
+            self[NameObject(key)] = BooleanObject(value)
```

Comment:
Why did you introduce those `_set_X` methods?

It seems way simpler to me to just access the dictionary attribute directly. Is there an adavantage of using the setter?

The behavior that setting a value of none actually deletes the key seems counter-intuitive to me.

Reply:
I will use these functions manytimes It will be simpler that way (see next commit)

Reply:
If you want a way to avoid using `NameObject` we could add

```
def _set(self, key: str, value: PdfObject) -> None:
    self[NameObject(key)] = value
```